### PR TITLE
Access refactored to enabled more types

### PR DESCRIPTION
**Pull request by dvoytenko**
Partial for #2970.

### CODE REVIEW

File: extensions/amp-access/0.1/amp-access-client.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    this.xhr_ = xhrFor(win);
+
+    /** @const @private {!Timer} */
+    this.timer_ = timer;
+  }
+
+  /** @override */
+  getConfig() {
+    return {
+      authorizationUrl: this.authorizationUrl_,
```
Comment (erwinmombay) [inline extensions/amp-access/0.1/amp-access-client.js]: do we need to put quotes on these keys for JSONObject? will more aggressive compilation level rename these keys?

File: extensions/amp-access/0.1/amp-access-other.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isProxyOrigin} from '../../../src/url';
+import {dev} from '../../../src/log';
```
Comment (erwinmombay) [inline extensions/amp-access/0.1/amp-access-other.js]: sort

File: extensions/amp-access/0.1/amp-access-client.js (hunk 1 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access-client.js]: Yeah. Good idea. Done.

File: extensions/amp-access/0.1/amp-access-other.js (hunk 2 above)
Comment (dvoytenko) [inline extensions/amp-access/0.1/amp-access-other.js]: Done

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

**Pull request merged**